vm/src/eval.rs

```diff
@@ -277,17 +286,65 @@ pub fn eval_inst(vm: &mut NexusVM<impl Memory>) -> Result<()> {
         ECALL { rd } => {
             RD = rd;
             vm.Z = vm.syscalls.syscall(vm.regs.pc, vm.regs.x, &vm.mem)?;
+
+            // Collects label from syscall and starts counting cycles
+            if vm.Z == (SyscallCode::ProfileCycles as u32) {
```

Comment:
Code-placement wise, it makes it such that this syscall implementation is part of vm, and others are in `Syscalls`. This functionality should be extracted and tested, including scenarios with bad input and not following the `^$` script.

Reply:
I agree with @slumber, but also agree redesigning the whole syscalls interface to support this would be a heavy lift at the moment. Maybe just leave a todo?

Reply:
I created an issue for this #264